### PR TITLE
TER-60 tests for typed table in Teradata

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: TER-60
Connection: https://github.com/keboola/connection/pull/4080

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

File: tests/Backend/Teradata/ImportTypedTableTest.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Keboola\Test\Backend\Teradata;
+
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Workspaces;
```
Comment (jirkasemmler) [inline tests/Backend/Teradata/ImportTypedTableTest.php:7]: tyto 3 tady nejsou potreba

File: tests/Backend/Teradata/ImportTypedTableTest.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+class ImportTypedTableTest extends ParallelWorkspacesTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $token = $this->_client->verifyToken();
+
+        if (!in_array('tables-definition', $token['owner']['features'])) {
```
Comment (jirkasemmler) [inline tests/Backend/Teradata/ImportTypedTableTest.php:20]: presne na tomto vyfailovaly vsechny testy v tagu . https://dev.azure.com/keboola-dev/26bf9dc4-aef5-4354-84fe-c52ee81d4cfa/_apis/build/builds/27939/logs/47 prosim pridej to do `ParatestBackendTeradataCmd` a otoc projekt na testingu

File: tests/Backend/Teradata/ImportTypedTableTest.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        $this->initEmptyTestBucketsForParallelTests();
+    }
+
+    public function testLoadTypedTablesConversionError(): void
+    {
+        $fullLoadFile = __DIR__ . '/../../_data/users.csv';
+
+        $bucketId = $this->getTestBucketId(self::STAGE_IN);
```
Comment (jirkasemmler) [inline tests/Backend/Teradata/ImportTypedTableTest.php:31]:
```suggestion
        $bucketId = $this->getTestBucketId();
```

File: tests/Backend/Teradata/ImportTypedTableTest.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        try {
+            // try import data with wrong types with full load
+            $this->_client->writeTableAsync(
+                $tableId,
+                new CsvFile($fullLoadFile),
+                [
+                    'incremental' => false,
+                ]
+            );
```
Comment (jirkasemmler) [inline tests/Backend/Teradata/ImportTypedTableTest.php:59]: nema tady byt fail? pokud jsi to kopiroval z jinych testu, tak to prosim dopln i tam

**Review by jirkasemmler — CHANGES_REQUESTED** [review]

File: tests/Backend/Teradata/ImportTypedTableTest.php (hunk 2 above)
Comment (martinjunger) [reply tests/Backend/Teradata/ImportTypedTableTest.php:20]: jj, vim, jdu na to

File: tests/Backend/Teradata/ImportTypedTableTest.php (hunk 3 above)
Comment (martinjunger) [reply tests/Backend/Teradata/ImportTypedTableTest.php:31]: je to takhle ve vsech testech, tak jsem to tam taky nechal :) Ale jo, muze to jit pryc.

File: tests/Backend/Teradata/ImportTypedTableTest.php (hunk 4 above)
Comment (martinjunger) [reply tests/Backend/Teradata/ImportTypedTableTest.php:59]: jo, dobrej point. doplneno i do ostatnich

**Comment by martinjunger** [comment]
Fixnul jsem spadly snapshot test a trochu upravil vytvareni defaultni tabulky (vytvarela se vzdycky, ale byla potreba jen v par testech).

**Review by jirkasemmler — APPROVED** [review]
cajk 🚀

**Pull request merged**